crates/database/src/settlement_observations.rs

```diff
@@ -0,0 +1,117 @@
+use {crate::events::EventIndex, bigdecimal::BigDecimal, sqlx::PgConnection};
+
+pub async fn get_settlement_event_without_observation(
+    ex: &mut PgConnection,
+    max_block_number: i64,
+) -> Result<Option<EventIndex>, sqlx::Error> {
+    const QUERY: &str = r#"
+SELECT block_number, log_index
+FROM settlement_observations
+WHERE gas_used IS NULL AND block_number <= $1
+LIMIT 1
```

Comment:
I don't know how this function is going to be used but it seems like you need an index on `(block_number, gas_used)` in order to make this query efficient.

Reply:
Function not used anymore.